systemtest/src/test/java/io/strimzi/systemtest/report/ClusterReportST.java

```diff
@@ -0,0 +1,528 @@
+            "--cluster=" + testStorage.getClusterName(),
+            "--out-dir=" + outPath,
+            "--secrets=off");
+
+        assertThat("Secrets are reported", !FileUtils.exists(secretPath));
+    }
+
+    @ParallelTest
+    void createReportWithoutNsOpts(TestInfo testInfo) {
+        LOGGER.info("Running report on {}/{}", testStorage.getNamespaceName(), testStorage.getClusterName());
+        assertThrows(KubeClusterException.class, () -> Exec.exec(USER_PATH + "/../tools/report.sh", "--cluster=" + testStorage.getClusterName()));
+    }
+
+    @ParallelTest
+    void createReportWithoutClusterOpts() {
+        LOGGER.info("Running report on {}/{}", testStorage.getNamespaceName(), testStorage.getClusterName());
+        assertThrows(KubeClusterException.class, () -> Exec.exec(USER_PATH + "/../tools/report.sh", "--namespace=" + testStorage.getNamespaceName()));
+    }
+
+    @ParallelTest
+    void createReportWithNonExistingNs() {
+        LOGGER.info("Running report on {}/{}", testStorage.getNamespaceName(), testStorage.getClusterName());
+        assertThrows(KubeClusterException.class, () -> Exec.exec(USER_PATH + "/../tools/report.sh",
+            "--namespace=non-existing-ns", "--cluster=" + testStorage.getClusterName()));
+    }
+
+    @ParallelTest
+    void createReportWithNonExistingCluster() {
+        LOGGER.info("Running report on {}/{}", testStorage.getNamespaceName(), testStorage.getClusterName());
+        assertThrows(KubeClusterException.class, () -> Exec.exec(USER_PATH + "/../tools/report.sh",
+            "--namespace=" + testStorage.getNamespaceName(), "--cluster=non-existing-k"));
+    }
+
+    @ParallelTest
+    void createReportWithUnknownSecretVerbosity() {
+        LOGGER.info("Running report on {}/{}", testStorage.getNamespaceName(), testStorage.getClusterName());
+        assertThrows(KubeClusterException.class, () -> Exec.exec(USER_PATH + "/../tools/report.sh",
+            "--namespace=" + testStorage.getNamespaceName(), "--cluster=" + testStorage.getClusterName(), "--secrets=unknown"));
+    }
+
+    @BeforeAll
+    void setUp(final ExtensionContext extensionContext) {
+        this.clusterOperator = this.clusterOperator
+            .defaultInstallation(extensionContext)
+            .withReplicas(2)
+            .withExtraEnvVars(Arrays.asList(new EnvVar(Environment.STRIMZI_FEATURE_GATES_ENV, "+KafkaNodePools,+StableConnectIdentities", null)))
+            .createInstallation()
+            .runInstallation();
+        this.testStorage = new TestStorage(extensionContext);
+        Kafka kafkaMainCr = KafkaTemplates.kafkaWithCruiseControl(testStorage.getClusterName(), 3, 3)
+            .editOrNewMetadata()
+                .addToAnnotations(Annotations.ANNO_STRIMZI_IO_NODE_POOLS, "enabled")
+            .endMetadata()
+            .build();
+        KafkaNodePool kafkaNodePoolCr =  KafkaNodePoolTemplates
+                .defaultKafkaNodePool("kafka", testStorage.getClusterName(), 3)
```

Comment:
If you wanna test how it works with node pools, you should probably create multiple non-default node pools. But then you likely need to also test it without the node pools to make sure it works without it as well.

Reply:
Yeah, that was a compromise to be more future proof, but we can cover both.